### PR TITLE
assignment 3

### PR DESCRIPTION
**Pull request by bapandaOK44**
trigrams.py; I tested it mostly with sherlock_small.txt; execute with $ trigrams.py sherlock_small.txt

### CODE REVIEW

File: students/barb_matthews/lesson4/trigrams.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+#! /usr/bin/env python3
+## Lesson 4 Assignment 3, Trigrams Kata 14
+## By: B. Matthews
+## https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/kata_fourteen.html
+## Note: this somewhat works, but could be better. I'm not sure how to keep the new
+## text going after it doesn't find a key match in the dictionary
+
+import os
```
Comment (charlesrobison) [inline students/barb_matthews/lesson4/trigrams.py:8]: Imported but not used

File: students/barb_matthews/lesson4/trigrams.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import sys
+import string
```
Comment (charlesrobison) [inline students/barb_matthews/lesson4/trigrams.py:10]: This one isn't used as well

File: students/barb_matthews/lesson4/trigrams.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    trigrams = {}
+
+    for i in range(len(w)-2):
+        key_words = tuple(w[i:i + 2])
+        third_word = w[i + 2]
+
+        if key_words in trigrams:
+            trigrams[key_words].append(third_word)
+        else:
+            trigrams[key_words]=[third_word]
```
Comment (charlesrobison) [inline students/barb_matthews/lesson4/trigrams.py:50]: Missing spaces around assignment operator

**Review by charlesrobison — APPROVED** [review]
Not quite getting the right output here:

('been', 'in') ['part']
('in', 'part') ['my']
('part', 'my') ['own.']
I rang  the bell and was shown up to the chamber which had formerly been in part my own.

Final output is the final line in sherlock_small.txt

Take another look and let us know if you need help.

Ready to merge

**Comment by bapandaOK44** [comment]
Hey Charles,

Ok. Try this version now. 

https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/1250 <https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/1250>

**Pull request merged**